dbt/parser.py

```diff
@@ -341,57 +304,52 @@ def parse_schema_tests(tests, root_project, projects):
     return to_return
 
 
-def parse_schema_test(test_base, model_name, test_config, test_type,
-                      root_project_config, package_project_config,
-                      all_projects):
-    if test_type == 'not_null':
-        raw_sql = QUERY_VALIDATE_NOT_NULL.format(
-            ref="{{ref('"+model_name+"')}}", field=test_config)
-        name_key = test_config
+def get_nice_schema_test_name(test_type, test_name, args):
 
-    elif test_type == 'unique':
-        raw_sql = QUERY_VALIDATE_UNIQUE.format(
-            ref="{{ref('"+model_name+"')}}", field=test_config)
-        name_key = test_config
+    flat_args = []
+    for arg_name in sorted(args):
+        arg_val = args[arg_name]
 
-    elif test_type == 'relationships':
-        if not isinstance(test_config, dict):
-            return None
+        if type(arg_val) == dict:
+            parts = arg_val.values()
+        elif type(arg_val) in (list, tuple):
+            parts = arg_val
+        else:
+            parts = [arg_val]
```

Comment:
can this be anything but a dict? looks like below we make sure it's a dict before calling this. `parts = kwargs.values()` should cover all the cases, right?